TST Incorporate global_random_seed in test_optics.py

simarssidhu opened the pull request:
### Changes
- updated four functions in `test_optics.py` to incorporate `global_random_seed` ensuring seed invariance:
    - test_extract_xi( )
        - this one was the most challenging one to update, and I'm still uncertain about the solution. Since the random seed affect clustering, expected labels are no longer hardcoded. Instead, `adjusted_rand_score` is used to compare clustering results and determine if they are roughly equivalent. However, it doesn't work for all seeds, so `global_random_seed % 10` was introduced. I was wondering if someone can steer me in the right direction for this problem. Thanks!
    - test_cluster_hierarchy_( )
        - increased `xi`, and assertion tolerance to accommodate randomness
    - test_extract_dbscan( )
        - modified assertion to handle outliers/noises
    - test_optics_input_not_modified_precomputed_sparse_nodiag( )
        - no functional changes, only incorporated `global_random_seed`

### Purpose:
- improved test robustness by handling randomness more effectively

### Related Issues:
- Helps solve #22827

simarssidhu committed:
Updated test_cluster_hierarchy to incorporate global_random_seed. Increased xi and assertion tolerance to include random seeds

simarssidhu committed:
Updated test_extract_dbscan to incorporate global_random_seed. Modified assertion to remove outlier/noise

simarssidhu committed:
Updated test_optics_input_not_modified_precomputed_sparse_nodiag to incorporate global_random_seed.

simarssidhu committed:
Updated test_extract_xi to incorporate global_random_seed. Modified assertions to account for cluster changes by random seeds

simarssidhu committed:
Checking on all runners [all random seeds]

simarssidhu commented:
Hi @ogrisel,

I was wondering, does this PR require a changelog? The change seems quite small, so I'd like to clarify if it may not need it.

Thanks,
Simar

OmarManzoor commented on sklearn/cluster/tests/test_optics.py:
```diff
@@ -81,10 +82,10 @@ def test_the_extract_xi_labels(ordering, clusters, expected):
 
 
-def test_extract_xi(global_dtype):
+def test_extract_xi(global_dtype, global_random_seed):
```
Maybe we should not add the global_random_seed for this particular test as it's quite dependent on the initial points and labels. What do you think @adrinjalali

simarssidhu replied:
Hi @OmarManzoor and @adrinjalali,

Thanks for the suggestion. The `test_extract_xi( )` function is quite dependent on the initial points and labels. I've reverted them for now. I've also rebased to main, and added an empty merge commit based on @ArturoSbr's [comment](https://github.com/scikit-learn/scikit-learn/issues/22827#issuecomment-2689380350). I was wondering if the rest is okay and ready to merge?

Thanks,
Simar

OmarManzoor reviewed (commented):
Thanks for the PR @simarssidhu

simarssidhu committed:
Revert "Updated test_extract_xi to incorporate global_random_seed. Modified assertions to account for cluster changes by random seeds"

This reverts commit f95acbf5ecb9b55f668eca0bd8ccacb361f6b6b8.

simarssidhu committed:
Merge branch 'main' into svm_optics_global_random_seed

simarssidhu committed:
empty commit to trigger all CI jobs [all random seeds] test_parallel

simarssidhu commented:
Thanks for adding the `No Changelog Needed` tag @StefanieSenger!

simarssidhu committed:
Merge branch 'main' into svm_optics_global_random_seed

simarssidhu commented:
Hi @ogrisel, @OmarManzoor, and @adrinjalali,

I was wondering if there is possible updates or further suggestions for this PR?

Thanks,
Simar

simarssidhu committed:
Merge branch 'main' into svm_optics_global_random_seed

jeremiedbb committed:
Merge remote-tracking branch 'upstream/main' into pr/simarssidhu/30844

jeremiedbb committed:
adjust test [all random seeds]
test_cluster_hierarchy
test_extract_dbscan
test_optics_input_not_modified_precomputed_sparse_nodiag

jeremiedbb reviewed (approved):
LGTM. Thanks @simarssidhu
